### PR TITLE
Update to a new analyzer package

### PR DESCRIPTION
**Pull request by pakrym**
It includes double newline and whitespace before line end analyzers.

### CODE REVIEW

**Review by tg-msft — APPROVED** [review]
It's going to be a blood bath of a diff when we turn these on for Storage...  I'll aim to do that sooner than later.

**Comment by pakrym** [comment]
> It's going to be a blood bath of a diff when we turn these on for Storage... I'll aim to do that sooner than later.

I'll do it as part of this PR.

**Comment by tg-msft** [comment]
So much for all the conspiracy theories I encoded via whitespace stenography in the Storage source...

but thanks for enabling them because now's a great time to make the change.  I filed #6947 to turn the rest of them on.

File: eng/Packages.Data.props [hunk 1]
```diff
@@ -9,7 +9,7 @@
   <ItemGroup>
     <PackageReference Update="ApprovalTests" Version="3.0.22" />
     <PackageReference Update="ApprovalUtilities" Version="3.0.22" />
-    <PackageReference Update="Azure.ClientSdk.Analyzers" Version="0.1.1-dev.20190711.4" />
+    <PackageReference Update="Azure.ClientSdk.Analyzers" Version="0.1.1-dev.20190716.1" />
```
Comment (weshaggard) [inline eng/Packages.Data.props]: nit: robably worth putting this package in the IsClientLibrary section of this file as well.

**Pull request merged**